components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -24,6 +24,27 @@ import mozilla.components.service.glean.private.CounterMetricType
 import mozilla.components.service.glean.private.TimingDistributionMetricType
 import mozilla.components.service.glean.private.LabeledMetricType
 
+/**
+ * This component can emit metrics via Glean, but the consuming application
+ * needs to explicitly opt-in. It does so by calling this function to tell
+ * us what version of our metrics it wants to opt in to.
+ *
+ * Thus, it's very important that when you add new metrics-collection in this
+ * component, you increment the value of `LATEST_METRICS_VERSION` to indicate it.
+ */
+
+private const val LATEST_METRICS_VERSION = 1
+private const val METRICS_V1 = 1
+private var enabledMetricsVersion = 0
```

Comment:
This should be a `@Volatile`, and you should consider synchronizing on something (at the moment synchronizing wouldn't matter, but it would be required to prevent stuff like TOCTOU errors if `enableTelemetry` grows to look at `enabledMetricsVersion` too).